### PR TITLE
docs(start_planner): update Purpose / Role of the document

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

update Purpose / Role of the document

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_start_planner_module/README.md [hunk 1]
```diff
@@ -2,17 +2,23 @@
 
 ## Purpose / Role
 
-The Start Planner module is designed to generate a path from the current ego position to the driving lane, avoiding static obstacles and implementing safety checks against dynamic obstacles. (Note: The feature of safety checks against dynamic obstacles is currently a work in progress.)
-This module is activated when a new route is received.
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/README.md:6]: This is not needed?

Comment (kyoichi-sugahara) [reply planning/behavior_path_start_planner_module/README.md:6]: changed expression in this PR from
```
The Start Planner module is designed to generate a path from the current ego position to the driving lane, avoiding static obstacles and implementing safety checks against dynamic obstacles. (Note: The feature of safety checks against dynamic obstacles is currently a work in progress.)
```
to
```
The Start Planner module is designed to generate a path from the current ego position to the driving lane, avoiding static obstacles and stopping in response to dynamic obstacles when a collision is detected.
```

And for the following description is not correct in my understanding :thinking: 
```
This module is activated when a new route is received.
```
activated condition is not only receiving new route?

Comment (kosuke55) [reply planning/behavior_path_start_planner_module/README.md:6]: I thought the tivated condition is only receiving new route. What are other conditions?
(I can not check the code details now🙇)

**Review by kosuke55 — APPROVED** [review]

File: planning/behavior_path_start_planner_module/README.md (hunk 1 above)
Comment (kosuke55) [reply planning/behavior_path_start_planner_module/README.md:6]: memo:
ex) even if reciving new route, the ego deviates from center line the module is not executed.

Comment (kosuke55) [reply planning/behavior_path_start_planner_module/README.md:6]: will update to add acivation condition section

**Pull request merged**